### ISSUE TITLE
Find returns no record for many to many relation

### ISSUE THREAD

**Issue by mahadazad** [balderdashy/sails-mongo#316]
I have following relations defined:

Post.js:
----------
```
module.exports = {
  connection: 'mongodb',

  attributes: {
    title: 'string',
    categories: {
      collection: 'postCategory',
      via: 'posts'
    }
  }
};
```

PostCategory.js
---------------------
```
module.exports = {
  connection: 'mongodb',

  attributes: {
    name: 'string',
    posts: {
      collection: 'post',
      via: 'categories'
    }
  }
};
```

When I query without criteria and populate `categories` like:
```
Post.find()
       .populate('categories')
       .then(...)
```

it gives me correct result, the document has categories nested.

But when I try to pass criteria it returns no result. e.g.

```
Post.find({categories: 'food'})
       .populate('categories')
       .then(...)
```

Note: I inserted category `_id` as string (food, travel, etc) in database

Please help

**Comment by Josebaseba** [comment]
Try this:

```javascript
Post.find()
       .populate('categories', {name: 'food'})
       .then(...)
```

**Comment by mahadazad** [comment]
@Josebaseba nope, its giving results irrelevant of category `food`

**Comment by devinivy** [comment]
Can you try this on the newly released 0.11.3?

**Comment by sailsbot** [comment]
Thanks for posting, @mahadazad.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
